rust/chains/tw_polkadot/src/extrinsic.rs

```diff
@@ -0,0 +1,554 @@
+use std::collections::HashMap;
+use std::convert::identity;
+use std::iter::{repeat, Iterator};
+use std::str::FromStr;
+
+use lazy_static::lazy_static;
+
+use tw_number::U256;
+use tw_proto::Polkadot::Proto;
+use tw_proto::Polkadot::Proto::mod_Balance::{
+    AssetTransfer, BatchAssetTransfer, BatchTransfer, OneOfmessage_oneof as BalanceVariant,
+    Transfer,
+};
+use tw_proto::Polkadot::Proto::mod_CallIndices::OneOfvariant as CallIndicesVariant;
+use tw_proto::Polkadot::Proto::mod_Identity::{
+    AddAuthorization, JoinIdentityAsKey, OneOfmessage_oneof as PolymeshIdentityVariant,
+};
+use tw_proto::Polkadot::Proto::mod_PolymeshCall::OneOfmessage_oneof as PolymeshVariant;
+use tw_proto::Polkadot::Proto::mod_SigningInput::OneOfmessage_oneof as SigningVariant;
+use tw_proto::Polkadot::Proto::mod_Staking::{
+    Bond, BondAndNominate, BondExtra, Chill, ChillAndUnbond, Nominate,
+    OneOfmessage_oneof as StakingVariant, Rebond, Unbond, WithdrawUnbonded,
+};
+use tw_proto::Polkadot::Proto::{Balance, PolymeshCall, Staking};
+use tw_ss58_address::{NetworkId, SS58Address};
+
+use crate::scale::{Compact, Raw, RawIter, ToScale};
+
+const POLKADOT_MULTI_ADDRESS_SPEC: u32 = 28;
+const KUSAMA_MULTI_ADDRESS_SPEC: u32 = 2028;
+
+// Common calls
+const BALANCE_TRANSFER: &str = "Balances.transfer";
+const STAKING_BOND: &str = "Staking.bond";
+const STAKING_BOND_EXTRA: &str = "Staking.bond_extra";
+const STAKING_CHILL: &str = "Staking.chill";
+const STAKING_NOMINATE: &str = "Staking.nominate";
+const STAKING_REBOND: &str = "Staking.rebond";
+const STAKING_UNBOND: &str = "Staking.unbond";
+const STAKING_WITHDRAW_UNBONDED: &str = "Staking.withdraw_unbonded";
+const UTILITY_BATCH: &str = "Utility.batch_all";
+
+// Non-existent calls on Polkadot and Kusama chains
+const ASSETS_TRANSFER: &str = "Assets.transfer";
+const JOIN_IDENTITY_AS_KEY: &str = "Identity.join_identity_as_key";
+const IDENTITY_ADD_AUTHORIZATION: &str = "Identity.add_authorization";
+
+type CallIndex = (u8, u8);
+type CallIndicesTable = HashMap<&'static str, CallIndex>;
+
+macro_rules! call_indices {
+    ($($chain:expr => { $($name:expr => ($($value:expr),*) $(,)?)* } $(,)? )*) => {
+        [
+            $((
+                $chain, std::collections::HashMap::from_iter(
+                    [$(($name, ($($value as u8),+))),+]
+                )
+            )),+
+        ].into_iter().collect()
+    }
+}
+
+lazy_static! {
+    static ref CALL_INDICES_BY_NETWORK: HashMap<NetworkId, CallIndicesTable> = call_indices! {
```

Comment:
I think it's worth to move the "standard" call indices to a dedicated module like `src/extrinsic/standard_extrinsic.rs` as well as `encode_transfer`, `encode_asset_transfer` etc, so to make `src/extrinsic/mod.rs` more general and re-usable.